### PR TITLE
Handle the error from new RootCoord for DescribeCollection

### PR DESCRIPTION
**Pull request by yah01**
/kind improvement
the old version Proxy may receive a response from a newer RootCoord, as we have modified the error code the `DescribeCollection`, and the SDK impl depends on the unexpected code and error message, this would happen during the rolling upgrade
pr: #26521

### CODE REVIEW

File: internal/proxy/task.go [hunk 1]
```diff
@@ -531,6 +523,8 @@ func (dct *describeCollectionTask) Execute(ctx context.Context) error {
 
 	if result.Status.ErrorCode != commonpb.ErrorCode_Success {
```
Comment (congqixia) [inline internal/proxy/task.go:524]: maybe add nil check for `result.Status`

Comment (yah01) [reply internal/proxy/task.go:524]: this is not related to this PR, will change in another PRs

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**